app/models/post_mover.rb

```diff
@@ -701,7 +701,9 @@ def close_topic_and_schedule_deletion
     @original_topic.update_status("closed", true, @user)
 
     days_to_deleting = SiteSetting.delete_merged_stub_topics_after_days
-    if days_to_deleting > 0
+    if days_to_deleting == -1
+      @original_topic.trash!
```

Comment:
It's probably a bad idea to use `.trash!` directly. IMO we should do the same as the timer-based action, which is here:

https://github.com/discourse/discourse/blob/f9b44361b26d6e135510aa27dd2e6894c97984a6/app/jobs/regular/delete_topic.rb#L6C13-L16C10

One important thing is that it logs the deletion to the staff-action-log.

Also, it also has an extra guardian check - not sure if we need that here as well.